src/MongoDB.Driver.Core/MongoException.cs

```diff
@@ -28,6 +30,9 @@ namespace MongoDB.Driver
 #endif
     public class MongoException : Exception
     {
+        // private fields
+        private readonly List<string> _errorLabels = new List<string>();
```

Comment:
It may not be necessary, but I used a Set to ensure no duplicates

Reply:
I changed AddErrorLabel to ensure no duplicates.

For small lists we can use List<string> instead of HashSet<string> and it will still be more efficient because the list is small. List<string> also preserves order (for what it's worth).

This is analogous to BsonDocument using a List<BsonElement> for documents of 8 or fewer elements.